dvc/objects/transfer.py

```diff
@@ -85,8 +133,10 @@ def transfer(odb, from_fs, from_info, jobs=None, no_progress_bar=False):
             from_fs,
             from_info,
             jobs=jobs,
+            update=known_hash and known_hash.isdir,
```

Comment:
Btw, `known_hash` pretty much doesn't matter at all here, as `save()` will call `odb.add()` that will check for existing object with the same object id (aka hash_info). So instead of passing `known_hash` to `transfer()` we coud've just passed `update=True/False`

Reply:
Agreed. It is a leftover from the previous logic and I kept it in case of might need it in the future though didn't remove. Will probably get rid of it when migrating to stage() (still thinking...  :eyes:)

Reply:
@isidentical Makes sense! Btw, this is one of the points of confusion that `object` refactor was intended to clear up. We used to toss around hash_infos when we really meant to use objects. So if we assume that `known_hash` was needed here, we should've used `known_object`(or old_obj) instead. Just illustrating the point :wink: , we are on the same page otherwise.